openfeature/provider.go

```diff
@@ -93,30 +95,93 @@ func (p *DatadogProvider) Metadata() openfeature.Metadata {
 
 // Init initializes the provider. For the Datadog provider,
 // this is waiting for the first configuration to be loaded.
-func (p *DatadogProvider) Init(openfeature.EvaluationContext) error {
+func (p *DatadogProvider) Init(evaluationContext openfeature.EvaluationContext) error {
+	// Use a background context with a reasonable timeout for backward compatibility
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```
Sadly I think we need to put at the very least 15sec or more because remote config can be really really slow